pkg/sdk/internal/collections/collection_helpers.go

```diff
@@ -0,0 +1,16 @@
+package collections
+
+import (
+	"errors"
+)
+
+var ErrObjectNotFound = errors.New("object does not exist")
+
+func FindOne[T any](collection []T, condition func(T) bool) (*T, error) {
```

Comment:
this feels like such a sad package. wondering what else we can put in here. Also does it make sense to have the error in this package included? then elsewhere it would have to be referenced as collections.ErrObjectNotFound

Reply:
Well, there may be much more convenience methods for collections there, and even if it stays "sad", I don't have a problem with it ;)

About errors, we discussed with @sfc-gh-jcieslak today; we were considering three options:
1. have this error here
2. move SDK errors to the internal package and use one of them here
3. change the signature of the method so that error type can be given to this function

We have decided to go with 1. However, based on your other comment above about moving errors to internal, I will go with 2.